### PR TITLE
docs: try to fit the doc with the code

### LINKED ISSUE #2538: definition for request hostname is incorrect
## 🐛 Bug Report

in the docs request `hostname` is defined as providing the hostname -- which is very incorrect for [what it actually does](https://github.com/fastify/fastify/blob/f2982a245e55f6c9d0a88837434e8423db6cc0ae/lib/request.js#L74-L81). 

what it actually does:
- if request.headers['x-forwared-for'] exists, the value is returned. (i believe this could be an array?)
- host header as-is (so it's returning "host" and not "hostname")
- if no host header exists, [return ":authority"](https://github.com/fastify/fastify/issues/2112)

so, the one thing request.hostname doesn't do is return the hostname.

### PR DESCRIPTION
**Pull request by rubiagatra**
Close #2538 

#### Checklist

- ~[ ] run `npm run test` and `npm run benchmark`~
- ~[ ] tests and/or benchmarks are included~
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Hello @mcollina, I just try to change documentation based on the issue, I don't know the words suit yet or not. Could you help me to review it?

Thanks!

### CODE REVIEW

File: docs/Request.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Request is a core Fastify object containing the following fields:
 - `log` - the logger instance of the incoming request
 - `ip` - the IP address of the incoming request
 - `ips` - an array of the IP addresses, ordered from closest to furthest, in the `X-Forwarded-For` header of the incoming request (only when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled)
-- `hostname` - the hostname of the incoming request (derived from `X-Forwarded-Host` header when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled)
+- `hostname` - the host of the incoming request (derived from `X-Forwarded-Host` header when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled). If no host header exist return `:authority`
```
Comment (mcollina) [inline docs/Request.md:17]:
```suggestion
- `hostname` - the host of the incoming request (derived from `X-Forwarded-Host` header when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled). For HTTP/2 compatibility it returns `:authority` if no host header exist.
```

**Review by mcollina — COMMENTED** [review]
A suggestion, feel free to rephrase.

**Comment by rubiagatra** [comment]
Thanks! I just applied your suggestion.

File: docs/Request.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Request is a core Fastify object containing the following fields:
 - `log` - the logger instance of the incoming request
 - `ip` - the IP address of the incoming request
 - `ips` - an array of the IP addresses, ordered from closest to furthest, in the `X-Forwarded-For` header of the incoming request (only when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled)
-- `hostname` - the hostname of the incoming request (derived from `X-Forwarded-Host` header when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled)
+- `hostname` - the host of the incoming request (derived from `X-Forwarded-Host` header when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled). For HTTP/2 compatibility it returns `:authority` if no host header exist.
```
Comment (darkgl0w) [inline docs/Request.md:17]:
```suggestion
- `hostname` - the host of the incoming request (derived from `X-Forwarded-Host` header when the [`trustProxy`](Server.md#factory-trust-proxy) option is enabled). For HTTP/2 compatibility it returns `:authority` if no host header exists.
```

Comment (darkgl0w) [reply docs/Request.md:17]: Just a little typo.

**Review by RafaelGSS — APPROVED** [review]
After fixing the above typo. LGTM.

**Comment by rubiagatra** [comment]
Yes, thank you @RafaelGSS @darkgl0w

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**